nflog/nflog.go

```diff
@@ -73,23 +74,19 @@ func QGroupKey(gk string) QueryParam {
 	}
 }
 
+// Log holds the notification log state for alerts that have been notified.
 type Log struct {
+	clock clock.Clock
```

Comment:
Most, if not all, of the diff in the tests, has to do with the fact that we now mock the clock instead of injecting. Technically, this can be done in separate PR, but this PR seems simple enough to follow so I decided to include it here.